plugins/civchat2-paper/src/main/java/vg/civcraft/mc/civchat2/CivChat2Manager.java

```diff
@@ -345,6 +381,40 @@ public void addGroupChat(Player player, Group group) {
         scoreboardHUD.updateScoreboardHUD(player);
     }
 
+
+    //method to check if a message contains a banned word
+    private boolean containsBannedWord(String message) {
+    String lowerMessage = message.toLowerCase();
+    
+    // Load banned words from resources
+    List<String> bannedWords = new ArrayList<>();
+    try (InputStream is = getClass().getResourceAsStream("/banned-words.txt");
+         BufferedReader reader = new BufferedReader(new InputStreamReader(is))) {
```

Comment:
Also you're reading the entire file through I/O on the main thread every time a message is sent which will have a severe performance penalty